tests/performance/plugins/server-timing/mu-plugin.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+add_action(
+	'template_redirect',
+	function() {
+		ob_start();
+
+		global $server_timing_values, $timestart;
+
+		if ( ! is_array( $server_timing_values ) ) {
+			$server_timing_values = array();
+		}
+
+		$server_timing_values['before-template'] = microtime( true ) - $timestart;
+
+		add_action(
+			'shutdown',
+			function() {
+
+				global $server_timing_values, $timestart;
+
+				if ( ! is_array( $server_timing_values ) ) {
+					$server_timing_values = array();
+				}
+
+				$server_timing_values['template'] = microtime( true ) - $timestart;
```

Comment:
This is slightly off, as right now it measures the total time. It needs to only measure the time starting from the above `template_redirect` hook.

You can define a variable like `$template_start` in that hook callback that stores the current `microtime( true )` and then use that in here instead of the `$timestart` global, which is only needed for the `before-template` metric.

Reply:
Thanks for the review and pointing it out. I have address this in [b2581a7](https://github.com/10up/wordpress-develop/pull/7/commits/b2581a70662d9917e23094359c0e2ce47dcd467d)